packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx

```diff
@@ -344,13 +350,17 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
           _.invoke(predefinedProps, 'onFocus', e, predefinedProps);
         },
         onClick: e => {
-          _.invoke(props, 'onDateChange', e, { ...props, value: day });
+          _.invoke(props, 'onDateChange', e, {
+            ...props,
+            value: day,
+            selectedDateRange: dateRangeType === DateRangeType.Week ? week : [day],
```

Comment:
might be better not to couple this to week
```suggestion
            selectedDateRange: dateRangeType !== DateRangeType.Day ? dateRange : [day],
```